### PR TITLE
linux: Enable Control Group feature on systemd environment

### PR DESCRIPTION
**Pull request by Ponsuke07**
# Purpose of pull request

Add support systemd of beaglebone black

# Test
## How to test

1. build beagleboneblack with systemd

```
MACHINE = "beaglebone"
DISTRO_FEATURES_append = " systemd"
DISTRO_FEATURES_BACKFILL_CONSIDERED += "sysvinit"
VIRTUAL-RUNTIME_init_manager = "systemd"
VIRTUAL-RUNTIME_initscripts = "systemd-compat-units"
```

2. Run beaglebone black

## Test result

If you do not have this fix, you will get the following error and failed to boot.

```
[    5.523808] systemd[1]: Failed to mount tmpfs at /sys/fs/cgroup: No such file or directory
[    5.540287] systemd[1]: Failed to mount cgroup at /sys/fs/cgroup/systemd: No such file or directory
[!!!!!!] Failed to mount API filesystems.
```

### CODE REVIEW

**Comment by masami256** [comment]
@Ponsuke07 Thank you for the patch !

I think cgroups isn't depend on devices, so it would be nice to move this setting from beaglebone.config to other file(e.g. resource-management.config). Then add following line(I haven't tested it) to linux-common.inc.

```
SRC_URI_append = " ${@oe.utils.conditional("VIRTUAL-RUNTIME_init_manager ", "systemd", " file://resource-management.config", "", d)}"
```

Or,  set new environment value to be able to use cgroups without systemd.

```
SRC_URI_append = " ${@oe.utils.conditional("SOME ENVIRONMENT VALUE NAME", "1", " file://resource-management.config", "", d)}"
```

Thereby user will have choice to use a cgroups or not.

**Comment by hiraku-wfs** [comment]
"systemd-dependencies.config" would be a better name.

**Comment by Ponsuke07** [comment]
@masami256 @hiraku-wfs Thank you for reply and suggestion.     
     
I have created v2 applying your points. 
**Note that** I only tested this with beaglebone.      
I think we should make sure that this patch is not adversely affecting other boards.  
\# The kernel-configs add by "+=" has a lower priority than "_append". 
\# So I believe my PR will not cause any major problems, but just in case

File: recipes-kernel/linux/files/systemd-dependencies.config [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+CONFIG_SIGNALFD=y
+CONFIG_TIMERFD=y
+CONFIG_EPOLL=y
+CONFIG_SYSFS=y
+CONFIG_PROC_FS=y
+CONFIG_FHANDLE=y
+CONFIG_CRYPTO_USER_API_HASH=m
+CONFIG_CRYPTO_HMAC=m
+CONFIG_CRYPTO_SHA256=m
+# CONFIG_SYSFS_DEPRECATED is not set
```
Comment (masami256) [inline recipes-kernel/linux/files/systemd-dependencies.config:14]: Do we need commented lines?

Comment (Ponsuke07) [reply recipes-kernel/linux/files/systemd-dependencies.config:14]: This is necessary to override config(=y) in the based defconfig when running [merge_config()](https://github.com/meta-debian/meta-debian/blob/warrior/classes/merge-config.bbclass#L31).

Comment (masami256) [reply recipes-kernel/linux/files/systemd-dependencies.config:14]: thanks! I got it.

**Comment by masami256** [comment]
I changed PR title because it doesn't depend on specific boards.

**Comment by masami256** [comment]
@Ponsuke07 I tested this commit on qemuarm64 and it works fine. Thanks!

**Comment by Ponsuke07** [comment]
Thank you so much!

**Pull request merged**